core/src/toga/app.py

```diff
@@ -449,15 +437,16 @@ def formal_name(self) -> str:
         return self._formal_name
 
     @property
-    def app_name(self) -> str:
+    def distribution_name(self) -> str:
         """The name of the distribution used to load metadata with
         :any:`importlib.metadata` (read-only)."""
-        return self._app_name
+        return self._distribution_name
```

Comment:
On reflection: I'm not 100% sure this deprecation/rename is warranted. 

The `name`->`formal_name` is definitely called for; `name` is ambiguous, especially given the usage in PEP621 context. 

However, I don't think renaming `app_name`->`distribution_name` is as much of a win.
* PEP621 talks about "distribution names" as a noun, but doesn't actually use `distribution_name` anywhere. There are APIs to `canonicalize_name()` and `normalize_name()`, which reference the `Name` metadata field. 
* Most of PEP621 is focussed on distributing libraries to PyPI, not "apps" in the Briefcase sense.
* The docs for `Name` call it the "distribution name"; I think you could reasonably argue that the "distribution" in any Toga/Briefcase context is an "app"
* Briefcase uses `app_name` extensively, so we'd be facing an inconsistency that we'd need to resolve.
* `app_name` isn't ambiguous, and is easy enough to document as "the distribution name".
* It's code churn for existing uses, and the new name is longer.

Reply:
OK, that makes sense; I'll revert the rename.

Things will be less confusing once we remove the deprecated `name` property, so I'll make an issue to remind us to do that.